search-scores-and-weights.md

```diff
@@ -0,0 +1,88 @@
+process. We also use scoring, which involves the assignment of numerical values
+and weights to documents based on various parameters, influencing their ranking
+in the search results. Here is our different scores, each serving a unique
+purpose:
+
+1. [**Similarity**](sql/2023-07-19-modify-score_type-add-similarity.sql):
+   Represents the primary signal in our scoring mechanism. It denotes the
+   measurement of how closely a document aligns with the user's query,
+   reflecting the relevance of the document to the search criteria. It is not a
+   static precomputed score but a dynamic metric that is computed on the fly
+   from the search results. This ensures that the relevance is tied to the
+   specific query, going beyond simple keyword matching. Documents with higher
+   similarity scores are considered more relevant. By prioritizing similarity as
+   the first signal in our scoring process, we aim to deliver search results
+   that are more accurate.
+      - Scaling: FROM 0.0 = least similar to user query TO 1.0 = most similar to
+        user query
+
+1. [**Recency**](sql/schema2.sql.public_new.sql): This score considers the
+   temporal aspect of documents, prioritizing recently added or updated content.
+   A document's recency is crucial in reflecting the latest information
+   available to users. 
+      - Scaling: FROM 0.0 = oldest document TO 1.0 = most recent document
+
+1. [**Traffic**](sql/compute-traffic-score.sql): The frequency with which users
+   consult a document influences its score. Popular or frequently accessed
+   documents are given higher scores with the help of web traffic logs,
+   indicating their relevance and importance to users. Warning: The home page is
+   rated really high since it's where every user land at first.
+      - Scaling: FROM 0.0 = least consulted document TO 1.0 = most consulted
+        document
+
+1. [**Current**](sql/2023-07-12-score-current.sql): This score determines
+   whether a document is currently accessible or if it has been archived. It
+   helps users distinguish between active and inactive content.
+      - Scaling: 0.0 = currently accessible document **OR** 1.0 = archived
+        document
+
+1. [**Typicality**](sql/2023-07-12-calculate-incoming-outgoing-counts.sql): This
+   score evaluates how closely the number of site references for a document
+   aligns with the average within a given thematic context. Documents with
+   typicality scores reflect a level of correspondence with the average number
+   of references, indicating their alignment within the specified topic. This
+   ensures that the search results prioritize documents not only based on their
+   relevance to the user's query but also considering how well they conform to
+   the typical reference patterns within the targeted theme.
+      - Scaling: FROM 0.0 = least referenced document TO 1.0 = most referenced
+        document
+
+1. **Didactic**: This score evaluates the informational value within content
```

Comment:
This score evaluates the informational value within content chunk. It scores higher based the quality and readability of information provided. Documents with high didactic scores often contain rich textual information, explanations, and details. Implementation of the score looks for signs of the opposite to compute its score - for example, the presence of a large proportion of tabular data which indicate data dumps from spreadsheets or databases.

Reply:
replaced